src/context/FilterProjectsContext.jsx

```diff
@@ -278,6 +286,15 @@ export const FilterProjectsProvider = ({ children }) => {
             records: project.records.filter((record) => isRecordWithinMapBounds(record, mapBbox)),
           }
         })
```

Comment:
The multiple chained and unnamed maps and filter on this data is intense. I know you are just adapting to an existing pattern here, but the whole thing could be better if each step is named properly/made self documenting. 

Not asking you to change it, just an observation! Im sure it got this way one isolated change at a time though. 

Anytime one is tempted to write a comment, its a queue to work on readability first. If that fails, _then_ write the comment, but not before attempting to make the code self documenting!

Reply:
the filter/map chain is definintely quite shocking when I see it. 
I often write comment where that part of code might be hard to read, but going with less code, and code readability is still my preferable way.